pkg/dynamicinstrumentation/codegen/codegen.go

```diff
@@ -0,0 +1,234 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package codegen is used to generate bpf program source code based on probe definitions
+package codegen
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+	"text/template"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/DataDog/datadog-agent/pkg/dynamicinstrumentation/ditypes"
+)
+
+// GenerateBPFParamsCode generates the source code associated with the probe and data
+// in it's associated process info.
+func GenerateBPFParamsCode(procInfo *ditypes.ProcessInfo, probe *ditypes.Probe) error {
+	parametersText := ""
```

Comment:
To prevent a bunch of extra allocations I might make this a `bytes.Buffer` and the pass it as an `io.Writer` to each of the `generate*` functions. That way they aren't each allocating strings, you don't need to create temporary `byte.Buffer` objects for the template `Execute` calls.

Reply:
Good idea, done.